src/pip/_internal/commands/wheel.py

```diff
@@ -161,10 +163,23 @@ def run(self, options, args):
                     build_options=options.build_options or [],
                     global_options=options.global_options or [],
                 )
-                build_failures = wb.build(
+                build_successes, build_failures = wb.build(
                     requirement_set.requirements.values(),
                     should_unpack=False,
                 )
+                for req in build_successes:
+                    assert req.link and req.link.is_wheel
+                    assert req.local_file_path
+                    # copy from cache to target directory
+                    try:
+                        ensure_dir(options.wheel_dir)
```

Comment:
I'd move `ensure_dir` out of this loop, perhaps in a followup. Note it is also done before in `Resolver.resolve`:

https://github.com/pypa/pip/blob/a4d06aecaa3f018e9604725798f2e8b9bb22024a/src/pip/_internal/legacy_resolve.py#L167-L168

Reply:
OK to do in a followup. If we move it to before the call to `resolve` above then no reason it cannot be removed from Resolver entirely later.